src/core/server/deprecations/README.mdx

```diff
@@ -0,0 +1,182 @@
+
+async function getDeprecations({ esClient, savedObjectsClient }: GetDeprecationsContext): Promise<DeprecationsDetails[]> {
+  const deprecations: DeprecationsDetails[] = [];
+  const testDashboardUser = await getTestDashboardUser(savedObjectsClient);
+
+  if (testDashboardUser) {
+  deprecations.push({
+    message: 'User "test_dashboard_user" is using a deprecated role: "kibana_user"',
+    documentationUrl: 'https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-user.html',
+    level: 'critical',
+    correctiveActions: {
+      api: {
+          path: '/internal/security/users/test_dashboard_user',
+          method: 'POST',
+          body: {
+            username: 'test_dashboard_user',
+            roles: [
+              'machine_learning_user',
+              'enrich_user',
+              'kibana_admin'
+            ],
+            full_name: 'Alison Goryachev',
+            email: 'alisongoryachev@gmail.com',
+            metadata: {},
+            enabled: true
+          }
+      },
+      manualSteps: [
+        'Using Kibana user management, change all users using the kibana_user role to the kibana_admin role.',
+        'Using Kibana role-mapping management, change all role-mappings which assing the kibana_user role to the kibana_admin role.'
+      ]
+    },
+  });
+  }
+
+  return deprecations;
+}
+ ```
+
+## How do I implement the API corrective action?
+The deprecations API allows plugins to provide an API call that can be used to automatically fix specific deprecations.
+To do so create a `PUT` or `POST` route in your plugin and specify data you want to be passed in the payload for the deprecation.
+
+In the example above `/internal/security/users/test_dashboard_user` will be called when users click on `Quick Resolve` in the UA. The service will automatically pass the body provided in the api corrective action to provide context to the route for fixing the deprecation.
+
+The deprecations service expects a `200` status code to recognize the corrective action as a success.
+
+## How do I test my deprecations?
+We recommend testing the route for your api corrective actions via unit and plugin functional tests.
+To check the deprecation in the UI pleae check the UA page.
+
+To test that your logic registering the deprecations with the deprecations service during `setup` is valid we recommend adding unit tests to check that the `getDeprecations` is returning the expected array of deprecations at different scenarios.
+
+Core provides mocks for the service which should help you focus on testing only your specific requirements.
+
+You can also use the deprecations service API; The deprecations service exposes an api `GET /api/deprecations/` which provides a list of deprecations and possible corrective actions required to resolve these entries. The context is scoped to the requesting user, hence a user with limited access might not be able to see all the deprecations affecting the deployment.
+
```

Comment:
Is there a way to test through the Upgrade Assistant?

Reply:
Ideally UA would add their test objects to allow functional testing through the UI. Currently the only way to test it is through what is mentioned above. CC @alisonelizabeth.